clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -2252,9 +2263,7 @@ public void handleResponse(AbstractResponse abstractResponse) {
                 // Check for controller change
                 for (ApiError error : response.errors().values()) {
                     if (error.error() == Errors.NOT_CONTROLLER) {
-                        metadataManager.clearController();
-                        metadataManager.requestUpdate();
-                        throw error.exception();
+                        handleNotControllerError(error.error());
```

Comment:
Not related

Reply:
Oops, this is a bug - it should be `handleNotControllerError(error)`